services/ui-src/src/components/fields/TextField.tsx

```diff
@@ -25,7 +26,9 @@ export const TextField = ({
     form.setValue(name, value, { shouldValidate: true });
   };
 
-  const errorMessage = form?.formState?.errors?.[name]?.message;
+  const errorMessage =
+    form?.formState?.errors?.[name]?.message ||
+    form?.formState?.errors?.[dynamic?.parentName]?.[dynamic?.index]?.message;
```

Comment:
I know I was involved in the creation of this monstrosity, but we should clean it up lol.

With a little work on the types (will have to define dynamic in component Props and guard against undefined) something like this should do the trick.

```suggestion
  const { parentName, index, inputRef } = dynamic;
  const formErrorState = form?.formState?.errors;
  const errorMessage =
    formErrorState?.[name]?.message ||
    formErrorState?.[parentName]?.[index]?.message;
```